agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/classicsdk/BytecodeUtilImpl.java

```diff
@@ -67,10 +69,15 @@ public void setConnectionString(String connectionString) {
               + " \"connectionStringConfiguredAtRuntime\" to true");
       return;
     }
-    if (TelemetryClient.getActive().getConnectionString() != null) {
-      logger.warn("Connection string is already set");
-      return;
+
+    if (!connectionStringProgrammaticallySet.getAndSet(true)
```

Comment:
maybe easier to read with something like `private static final AtomicBoolean firstCall = new AtomicBoolean(true);`
```suggestion
    if (firstCall.getAndSet(false)
```